test/SlowTests/Core/Streaming/QueryStreaming.cs

```diff
@@ -236,6 +237,71 @@ public void TestFailingProjection()
             }
         }
 
+        class TestObj
+        {
+            public string Id { get; set; }
+            public Dictionary<string, string> Prop { get; set; }
+        }
+        
+        [Fact]
+        public async Task QueryStream_WhenDocsContainsMultipleUniqPropertyNames_ShouldNotBeVeryVerySlow()
+        {
+            using var store = GetDocumentStore();
+
+            var objs = Enumerable.Range(0, 100000).Select(_ => new TestObj
+            {
+                Prop = new Dictionary<string, string>
+                {
+                    {Guid.NewGuid().ToString("n"), "someValue"},
+                    {Guid.NewGuid().ToString("n"), "someValue"},
+                    {Guid.NewGuid().ToString("n"), "someValue"}
+                }
+            }).ToArray();
+
```

Comment:
Can avoid materializing this by operating on the enumerable directly?

Reply:
I am not sure I understand your comment.
Do you want it without `ToArray`?

Reply:
Yes, then you can simple `foreach` on the values without creating a big array